engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/AncientDataMotionStrategy.java

```diff
@@ -587,6 +590,7 @@ protected Answer copySnapshot(DataObject srcData, DataObject destData) {
             if (cacheData != null) {
                 cacheMgr.deleteCacheObject(cacheData);
             }
+            _snapshotService.markFailedSnapshot(destData.getId());
```

Comment:
@Spaceman1984 the copy callback should mark the snapshot state accordingly, based on the operation result (success / fail). Not required to set the state explicitly.

Reply:
Addressed